### PR TITLE
Allow custom SSH user in UI for built-in SSH server (#2617)

### PR DESCRIPTION
**Pull request by pgaskin**
This allows a custom username to be set for the built-in SSH server.

It adds the config option `BUILTIN_SSH_USER` in the `[server]` section. If this option is missing or empty, it defaults to the `RUN_USER` (which was the default before), and it it is present and the built-in SSH server is enabled, then it uses the custom user for the built-in SSH server.

This does not introduce any compatibility issues with old versions.

I have tested this feature.

See issue #2617

### CODE REVIEW

**Comment by lafriks** [comment]
LGTM

**Comment by lunny** [comment]
It's not format well

**Comment by pgaskin** [comment]
@lunny You mean I need to run gofmt?

**Comment by daviian** [comment]
@geek1011 You can use `make fmt` too

**Comment by bkcsoft** [comment]
Shouldn't that user be verified in the builtin ssh-server as well?

**Comment by pgaskin** [comment]
@bkcsoft no, that would introduce compatibility issues and an unnecessary check. The SSH user for the built-in SSH server has no real meaning.

**Comment by pgaskin** [comment]
I did make fmt.

**Comment by pgaskin** [comment]
Can this be merged now?

File: modules/setting/setting.go
Comment (Morlinest) [inline modules/setting/setting.go]: Can this be changed to `SSH.BuiltinServerUser = sec.Key("BUILTIN_SSH_USER").MustString(RunUser)`?

File: modules/setting/setting.go
Comment (Morlinest) [inline modules/setting/setting.go]: Why setting default string value?

File: conf/app.ini
Comment (Morlinest) [inline conf/app.ini]: IMO is better to use name like `BUILTIN_SSH_SERVER_USER`.

**Review by Morlinest — CHANGES_REQUESTED** [review]
Some suggestions

File: modules/setting/setting.go
Comment (pgaskin) [reply modules/setting/setting.go]: I'll remove that.

File: modules/setting/setting.go
Comment (pgaskin) [reply modules/setting/setting.go]: It is set to the runuser if blank a bit further down. I cannot combine them because this one requires the server sec variable, and the runuser is set further down. (Line 922)

File: conf/app.ini
Comment (pgaskin) [reply conf/app.ini]: Should I change it?

Comment (pgaskin) [reply conf/app.ini]: I have changed it.

File: modules/setting/setting.go
Comment (Morlinest) [reply modules/setting/setting.go]: Yes, I saw that line. Can you move this setup line down as `BuiltinSSHServerUser` is not used between lines 725 and 922?

File: modules/setting/setting.go
Comment (sapk) [inline modules/setting/setting.go]: Couldn't we do simply SSH.BuiltinSSHServerUser = sec.Key("BUILTIN_SSH_USER").MustString(RunUser) here ?

Comment (Morlinest) [reply modules/setting/setting.go]: @sapk That's my question :D

Comment (sapk) [reply modules/setting/setting.go]: @Morlinest sorry I haven't look at others reviews. just quickly look at it ^^

Comment (pgaskin) [reply modules/setting/setting.go]: So my code is fine?

Comment (sapk) [reply modules/setting/setting.go]: What we suggest is that you move (and add the RunUser as default) L724 `SSH.BuiltinSSHServerUser = sec.Key("BUILTIN_SSH_USER").MustString(RunUser)` in place of L921-924

Comment (pgaskin) [reply modules/setting/setting.go]: @sapk It cannot happen because sec is redefined in between those spots. That's why I have the two lines.

Comment (geek1011) [reply modules/setting/setting.go]: @sapk It cannot happen because sec is redefined in between those spots. That's why I have the two lines.

Comment (Morlinest) [reply modules/setting/setting.go]: `Cfg.Section("server").Key("BUILTIN_SSH_SERVER_USER").MustString(RunUser)`?

Comment (pgaskin) [reply modules/setting/setting.go]: @Morlinest I didn't know about that one. Thanks. I'll try it.

**Comment by pgaskin** [comment]
@sapk @Morlinest I have updated according to your review.

**Review by lafriks — APPROVED** [review]

File: modules/setting/setting.go
Comment (Morlinest) [inline modules/setting/setting.go]: `BUILTIN_SSH_SERVER_USER`

**Review by Morlinest — CHANGES_REQUESTED** [review]

File: modules/setting/setting.go
Comment (Morlinest) [inline modules/setting/setting.go]: Idea: What about changing `BuiltinSSHServerUser` to `BuiltinServerUser`. Then reading config inside go `setting.SSH.BuiltinServerUser`. Also "builtin" word can be removed so it would be more consistent with other ssh settings. What do you think?

Comment (pgaskin) [reply modules/setting/setting.go]: I'll remove the SSH from the var name. Since this user only applies to the internal server, I'll keep the Builtin

File: modules/setting/setting.go
Comment (pgaskin) [reply modules/setting/setting.go]: Oops

**Comment by pgaskin** [comment]
@Morlinest should be fixed now.

File: modules/setting/setting.go
Comment (Morlinest) [reply modules/setting/setting.go]: Good point. The last thing: `SSH_BUILTIN_SERVER_USER` instead of `BUILTIN_SSH_SERVER_USER`?

**Comment by lafriks** [comment]
@Morlinest need your approval

**Comment by Morlinest** [comment]
@lafriks Waiting for response (last thing), I think `SSH_BUILTIN_SERVER_USER` would be better than `BUILTIN_SSH_SERVER_USER`.

**Comment by pgaskin** [comment]
@Morlinest how many times do you have to change your mind?

**Comment by lafriks** [comment]
@Morlinest I think current is better as SSH_BUILTIN_SERVER_USER does not sound correct in English

**Comment by Morlinest** [comment]
@lafriks I meant `SSH_` as prefix in config. In code it translates to `SSH.BuiltinServerUser`.

**Comment by Morlinest** [comment]
@geek1011 I was asking about another idea after accepting your opinion. I don't see anything bad on it.

**Comment by Morlinest** [comment]
@lafriks Same as `SSH_BACKUP_AUTHORIZED_KEYS` or `SSH_EXPOSE_ANONYMOUS`

**Comment by lafriks** [comment]
@Morlinest not really:
SSH_BUILTIN_SERVER_USER -> built-in server in ssh 
BUILTIN_SSH_SERVER_USER -> built-in ssh server

**Review by Morlinest — APPROVED** [review]

**Comment by Morlinest** [comment]
@lafriks OK, I accept your interpretation.

LGTM

**Comment by lafriks** [comment]
Make LG-TM work again

**Pull request merged**
